src/commands/aws/files.ts

```diff
@@ -0,0 +1,106 @@
+
+You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
+**/
+import { AwsCredentials } from "../../plugins/aws/types";
+import { EncodeOptions, parse, stringify } from "ini";
+import fs from "node:fs/promises";
+import os from "node:os";
+import path from "node:path";
+import tmp from "tmp-promise";
+
+const AWS_CONFIG_PATH = path.join(os.homedir(), ".aws");
+export const AWS_CONFIG_FILE = path.join(AWS_CONFIG_PATH, "config");
+export const AWS_CREDENTIALS_FILE = path.join(AWS_CONFIG_PATH, "credentials");
+
+// Reference documentation: https://docs.aws.amazon.com/sdkref/latest/guide/file-format.html
+
+/**
+ * Reads in an AWS CLI configuration file, which is formatted as INI text, and
+ * returns an arbitrary object representing the contents.
+ *
+ * @param path Path of the file to read
+ * @returns Arbitrary object representing the contents of the file, or an empty
+ * object if the file is empty or does not exist
+ */
+export const readIniFile = async (path: string): Promise<any> => {
+  let data;
+
+  try {
+    data = await fs.readFile(path, { encoding: "utf-8" });
+  } catch (err: any) {
+    if (err.code === "ENOENT") {
+      return {};
+    }
+
+    throw err;
+  }
+
+  if (!data) {
+    return {};
+  }
+
+  return parse(data);
+};
+
+/**
+ * This function writes an arbitrary object as INI-formatted text to a file
+ * atomically by first writing the data to a temporary file then moving the
+ * temporary file on top of the target file. This minimizes the chance that an
+ * exception, system crash, or other similar event will leave the file in a
+ * corrupted state; this is important since we're mucking around with the AWS
+ * CLI's configuration files.
+ *
+ * @param path Path of the (permanent) file to write to
+ * @param obj Arbitrary object to convert to INI-formatted text and write to the
+ * file
+ * @param iniEncodeOptions Options to pass to the INI encoding library
+ */
+export const atomicWriteIniFile = async (
+  path: string,
+  obj: any,
+  iniEncodeOptions?: EncodeOptions
+): Promise<void> => {
+  const data = stringify(obj, iniEncodeOptions);
+
+  // Permissions will be moved along with the file
+  const { path: tmpPath } = await tmp.file({ mode: 0o600, prefix: "p0cli-" });
+
+  await fs.writeFile(tmpPath, data, { encoding: "utf-8" });
+  await fs.rename(tmpPath, path);
+};
+
+export const writeAwsTempCredentials = async (
+  profileName: string,
+  awsCredentials: AwsCredentials
+) => {
+  const credentials = await readIniFile(AWS_CREDENTIALS_FILE);
+
+  credentials[profileName] = {
+    aws_access_key_id: awsCredentials.AWS_ACCESS_KEY_ID,
+    aws_secret_access_key: awsCredentials.AWS_SECRET_ACCESS_KEY,
+    aws_session_token: awsCredentials.AWS_SESSION_TOKEN,
```

Comment:
we also have to write the: `AWS_SECURITY_TOKEN`, it's a legacy value that gets used by other applications. it can be set to `AWS_SESSION_TOKEN` value.

https://github.com/p0-security/p0cli/blob/2ae3841b9ae920cbee4f0dd895964270af85ab31/src/plugins/aws/assumeRole.ts#L37

Reply:
`aws_security_token` is not part of the [documented spec of the AWS CLI configuration file format](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html#cli-configure-files-settings), and appears to have been deprecated in 2014 and no longer appears in any AWS documentation, aside from isolated callouts in the context of backwards compatibility. Indeed, AWS' [Ansible docs](https://docs.ansible.com/ansible/latest/collections/amazon/aws/sts_assume_role_module.html#parameters) call out that,

> Support for the `EC2_SECRET_KEY` and `AWS_SECURITY_TOKEN` environment variables has been deprecated and will be removed in a release after 2024-12-01.

I think it's safe to not write out this value.

Reply:
In practice @gergas3 and I found that this value is still being used in practice. When debugging SSH with Venkat we found that Okta sets the SECURITY_TOKEN value which broke our integration.

See description in: https://github.com/p0-security/p0cli/commit/a21a7b261624559c2b72275bc9841fc22b0cff72

Reply:
Hmm, so in looking into this further, I don't think the `AWS_SECURITY_TOKEN` environment variable will be an issue because we aren't setting any environment variables; we write the credentials to the AWS CLI configuration file in a profile and then pass that profile name on to subsequent commands, and that profile is what `kubectl` is using under the hood to get the EKS token. I've confirmed this with testing.

On the other hand, in the process of testing this, I discovered a separate potential issue—for some reason, even though we invoke `aws eks update-kubeconfig` with the profile passed in as a CLI argument (e.g., `--profile profile-name`), the kubectl configuration it writes passes the profile name itself as an environment variable (`AWS_PROFILE`). This means that if the `AWS_ACCESS_KEY_ID` environment variable is set (and only that one, because that's how AWS determines if there are credentials in environment variables) it will attempt to use environment variables instead of the profile. I'm going to push one more change to address this in a quick way, which is just to detect if the `AWS_ACCESS_KEY_ID` envvar is set and emit a warning if so.